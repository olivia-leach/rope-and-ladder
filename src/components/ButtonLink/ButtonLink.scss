//#d4ba8c
$orange: rgb(221, 125, 69);
$buttonColor: $orange;

.button-link {
  opacity: 1;

  &:hover {
    cursor: pointer;

    .button-link__div {
      color: #fff;

      &:after {
        transform: scaleX(1);
      }
    }
  }

  &__div {
    box-shadow: 0 0 0px rgba(0, 0, 0, 0);
    transition-property: 'color';
    transition-duration: 0.3s;
    border: 1px solid $buttonColor;
    color: $buttonColor;
    padding: 22px;
    width: 195px;
    text-align: center;
    align-content: center;
    text-transform: uppercase;
    font-weight: 900;
    letter-spacing: 1.5px;
    position: relative;
    text-transform: uppercase;

    &:after {
      content: '';
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      background: $buttonColor;
      transform: scaleX(0);
      transform-origin: 50%;
      transition-property: transform;
      transition-duration: 0.2s;
      transition-timing-function: ease-out;
    }
  }

  &__span {
    z-index: 1;
    position: absolute;
    top: 13px;
    bottom: 0;
    left: 0;
    right: 0;
    font-size: 0.9rem;
  }

  &--filled {
    .button-link__div {
      background-color: $buttonColor;
      color: #fff;

      &:after {
        background: #fff;
      }
    }

    &:hover {
      .button-link__div {
        color: $buttonColor;
      }
    }
  }
}
