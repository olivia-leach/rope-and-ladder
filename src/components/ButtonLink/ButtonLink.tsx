import React from 'react'
import './ButtonLink.scss'

interface IButtonLinkProps {
  icon: string
  label: string
  link: string
  filled?: boolean
  suppressNewPage?: boolean
}

const ButtonLink: React.FC<IButtonLinkProps> = ({
  icon,
  label,
  link,
  filled,
  suppressNewPage,
}) => (
  <a
    data-testid='buttonLink'
    href={link}
    target={suppressNewPage ? undefined : '_blank'}
    className={`button-link ${filled ? 'button-link--filled' : ''}`}
  >
    <div className="button-link__div">
      <span className="button-link__span">
        <i className={icon}></i> {label}
      </span>
    </div>
  </a>
)

export default ButtonLink
