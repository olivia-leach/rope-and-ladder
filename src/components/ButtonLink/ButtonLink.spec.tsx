import { render } from '@testing-library/react'
import React from 'react'
import ButtonLink from './ButtonLink'

const defaultProps = {
  icon: 'fa fa-shopping-cart',
  label: 'hi there',
  link: 'http://muffinfilms.com/',
  filled: true,
  suppressNewPage: false,
}

describe('ButtonLink', () => {
  it('should match snapshot when filled', () => {
    const { container } = render(<ButtonLink {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when not filled', () => {
    const { container } = render(<ButtonLink {...defaultProps} filled={false} />)
    expect(container).toMatchSnapshot()
  })

  it('should open links in new tab if suppressNewPage prop is false', () => {
    const { getByTestId, container } = render(<ButtonLink {...defaultProps} />)
    expect(container).toMatchSnapshot()
    expect(getByTestId('buttonLink')).toHaveProperty('target', '_blank')
  })

  it('should open links in same tab if suppressNewPage prop is true', () => {
    const { getByTestId, container } = render(<ButtonLink {...defaultProps} suppressNewPage />)
    expect(container).toMatchSnapshot()
    expect(getByTestId('buttonLink')).toHaveProperty('target', '')
  })
})
