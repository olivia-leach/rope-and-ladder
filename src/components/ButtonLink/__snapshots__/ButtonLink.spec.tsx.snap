// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should match snapshot when filled 1`] = `
<div>
  <a
    class="button-link button-link--filled"
    data-testid="buttonLink"
    href="http://muffinfilms.com/"
    target="_blank"
  >
    <div
      class="button-link__div"
    >
      <span
        class="button-link__span"
      >
        <i
          class="fa fa-shopping-cart"
        />
         
        hi there
      </span>
    </div>
  </a>
</div>
`;

exports[`ButtonLink should match snapshot when not filled 1`] = `
<div>
  <a
    class="button-link "
    data-testid="buttonLink"
    href="http://muffinfilms.com/"
    target="_blank"
  >
    <div
      class="button-link__div"
    >
      <span
        class="button-link__span"
      >
        <i
          class="fa fa-shopping-cart"
        />
         
        hi there
      </span>
    </div>
  </a>
</div>
`;

exports[`ButtonLink should open links in new tab if suppressNewPage prop is false 1`] = `
<div>
  <a
    class="button-link button-link--filled"
    data-testid="buttonLink"
    href="http://muffinfilms.com/"
    target="_blank"
  >
    <div
      class="button-link__div"
    >
      <span
        class="button-link__span"
      >
        <i
          class="fa fa-shopping-cart"
        />
         
        hi there
      </span>
    </div>
  </a>
</div>
`;

exports[`ButtonLink should open links in same tab if suppressNewPage prop is true 1`] = `
<div>
  <a
    class="button-link button-link--filled"
    data-testid="buttonLink"
    href="http://muffinfilms.com/"
  >
    <div
      class="button-link__div"
    >
      <span
        class="button-link__span"
      >
        <i
          class="fa fa-shopping-cart"
        />
         
        hi there
      </span>
    </div>
  </a>
</div>
`;
